### ISSUE TITLE
Missing registrant attribute support in signature documents

### ISSUE THREAD

**Issue by robotman3000** [SparkDevNetwork/Rock#5518]
### Please go through all the tasks below

- [X] Check this box only after you have successfully completed both the above tasks

### Please provide a brief description of the problem. Please do not forget to attach the relevant screenshots from your side.

When using the signature document feature of the new obsidian registration block, phone numbers and registrant attributes aren't supported in the signature document template.

### Expected Behavior

The template should have all values filled matching what was entered into the registration

### Actual Behavior

The template is missing any phone number in a person field and all registrant attributes

### Steps to Reproduce

 This issue was reproduced using the demo site.

1. Create a signature document using the following template
```
<p>Name: {{ Registrant.FirstName }} {{ Registrant.LastName }}</p>
<p>Phone: {{ Registrant.MobilePhone }}</p>
<p>Attribute: {{ Registrant | Attribute:"TestAttr" }}</p>
```
2. Create a registration with the following fields
![Screenshot 2023-07-17 at 4 19 25 PM](https://github.com/SparkDevNetwork/Rock/assets/10005584/7cb04bf4-e268-4cdd-ac45-725d986fc919)

3. Ensure you are using the obsidian registration entry block
4. Complete the registration
![Screenshot 2023-07-17 at 4 21 14 PM](https://github.com/SparkDevNetwork/Rock/assets/10005584/9eb7ec59-90ab-4a30-b55b-d3eb665e34e2)
---
![Screenshot 2023-07-17 at 4 21 28 PM](https://github.com/SparkDevNetwork/Rock/assets/10005584/55338576-1159-46fc-98a3-49576e0cdc0f)
---
[Alisha_Admin_(Signature_Document_Bug_Test_Instance).pdf](https://github.com/SparkDevNetwork/Rock/files/12073651/Alisha_Admin_.Signature_Document_Bug_Test_Instance.pdf)

### Rock Version

1.15.0.6

### Client Culture Setting

en-US

**Comment by frogydiak** [comment]
Yes, I was trying to create a registration form for an event with Signature Document and depending on the answer, the applicable section in the signature document will be included. For example: A Boolean field "I give my consent to authorize the church to use the photos/video of myself", if the registrant answered Yes, the Photo Release section will appear.

It turned out it is a bug. Thank you for fixing this @robotman3000, hopefully it will be included in 15.2.

**Comment by ethan-sparkdevnetwork** [comment]
- Cherry picked PR 5520 to branch hotfix 15.2: 19e6fb03
- Minor formatting changes: bba98dbd
- Refactor Lava merge fields for signature doc templates: 6240cc8e
- Updated template tips on SignatureDocumentTemplateDetail: ef43126a
